sdk/compiler/daml-lf-proto/test/DA/Daml/LF/Proto3/EncodeDecodeTest.hs

```diff
@@ -0,0 +1,131 @@
+-- Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Daml.LF.Proto3.EncodeDecodeTest (
+        module DA.Daml.LF.Proto3.EncodeDecodeTest
+) where
+
+
+import qualified Data.NameMap                             as NM
+
+import           DA.Daml.LF.Proto3.Encode
+import           DA.Daml.LF.Proto3.Decode
+
+import           DA.Daml.LF.Ast
+import qualified Com.Digitalasset.Daml.Lf.Archive.DamlLf2 as P
+
+import           Test.Tasty.HUnit
+import           Test.Tasty
+
+entry :: IO ()
+entry = defaultMain $ testGroup "All tests" [ rtt_tests ]
+
+------------------------------------------------------------------------
+-- Params
+------------------------------------------------------------------------
+testVersion :: Version
+testVersion = Version V2 PointDev
+
+------------------------------------------------------------------------
+-- Round-trip
+------------------------------------------------------------------------
+
+roundTripPackage :: Package -> Either Error Package
+roundTripPackage p = (decode . encodePackage) p
+  where
+    decode :: P.Package -> Either Error Package
+    decode = decodePackage
+      (packageLfVersion p) --from passed package
+      SelfPackageId
+
+
+roundTripAssert :: Package -> Assertion
+roundTripAssert p =
+  either
+    (\err -> assertFailure $ "Unexpected error: " ++ show err)
+    (\val -> p @=? val)
+    (roundTripPackage p)
+
+rtt_tests :: TestTree
+rtt_tests = testGroup "Round-trip tests"
+    [ rtt_empty
+    , rtt_tyLam
+    ]
+
+rtt_empty :: TestTree
+rtt_empty = testCase "empty package" $ roundTripAssert $ oneModulePackage emptyModule
+
+rtt_tyLam :: TestTree
+rtt_tyLam = testCase "tylam package" $ roundTripAssert $ oneModulePackage tyLamModule
+
+------------------------------------------------------------------------
+-- Examples
+------------------------------------------------------------------------
+emptyModule :: Module
+emptyModule = Module{..}
```

Comment:
This would be a nice utility to have everywhere, moving it to DA.Daml.LF.Ast.Util would be ideal